src/main/java/frc/team2412/robot/subsystem/ShooterSubsystem.java

```diff
@@ -1,11 +1,90 @@
 package frc.team2412.robot.subsystem;
 
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.TalonFXFeedbackDevice;
 import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
-
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 
 public class ShooterSubsystem extends SubsystemBase {
-    public ShooterSubsystem(WPI_TalonFX flywheelMotor1, WPI_TalonFX flywheelMotor2, WPI_TalonFX turretMotor, WPI_TalonFX hoodMotor) {
+    private final WPI_TalonFX flywheelMotor1;
+    private final WPI_TalonFX flywheelMotor2;
+    private final WPI_TalonFX turretMotor;
+    private final WPI_TalonFX hoodMotor;
+
+    public static final double MAX_FORWARD = 0.1;
+    public static final double MAX_REVERSE = -0.1;
+    public static final double STOP_MOTOR = 0;
+    public static final double DEGREES_TO_ENCODER_TICKS = 2048 / 360; // 2048 ticks per 360 degrees
+    public static final double MIN_TURRET_ANGLE = 0;
+    public static final double MAX_TURRET_ANGLE = 180; // Placeholder, maybe need to update
+
+    public ShooterSubsystem(WPI_TalonFX flywheelMotor1, WPI_TalonFX flywheelMotor2, WPI_TalonFX turretMotor,
+            WPI_TalonFX hoodMotor) {
+        this.flywheelMotor1 = flywheelMotor1;
+        this.flywheelMotor2 = flywheelMotor2;
+        this.flywheelMotor2.setInverted(true);
+        this.turretMotor = turretMotor;
+        this.hoodMotor = hoodMotor;
+        // Make sure hood is using builtin encoder, may be unneccessary
+        this.hoodMotor.configSelectedFeedbackSensor(TalonFXFeedbackDevice.IntegratedSensor, 0, 0);
+    }
+
+    public void hoodMotorExtend() {
+        hoodMotor.set(MAX_FORWARD);
+    }
+
+    public void hoodMotorRetract() {
+        hoodMotor.set(MAX_REVERSE);
+    }
+
+    public void hoodMotorStop() {
+        hoodMotor.set(STOP_MOTOR);
+    }
+
+    // TODO make sure these motors are going in the right direction
+    public void startFlywheel() {
+        flywheelMotor1.set(MAX_FORWARD);
+        flywheelMotor2.set(MAX_FORWARD);
+    }
+
+    public void stopFlywheel() {
+        flywheelMotor1.set(STOP_MOTOR);
+        flywheelMotor2.set(STOP_MOTOR);
+    }
+
+    /**
+     * Resets the turret motor's integrated encoder to 0.
+     */
+    public void resetTurretEncoder() {
+        turretMotor.setSelectedSensorPosition(0);
+    }
```

Comment:
I hoped we'd use an absolute encoder for the turret so that this isn't necessary. But if not, ideally this would auto-zero the turret based on some other signal (magnetic limit switch, vision, etc.).  Though until that's setup, I guess this is needed.

Reply:
Yes, we're planning on using a limit switch. Will add a TODO for now to be clear